platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -204,6 +205,8 @@ private void initialiseMap() {
       mapboxMap.onRestoreInstanceState(savedInstanceState);
     }
 
+    addDefaultRemoveUnusedStyleImagesListener();
```

Comment:
This call should be moved to the `MapView#initialize` method. Since `MapView#initialiseMap` is called only after the surface has been created, there can already be a listener registered by the user.